flake.nix

```diff
@@ -3,54 +3,102 @@
 
   inputs = {
     nixpkgs.url = "nixpkgs/nixpkgs-unstable";
-    utils.url = "github:numtide/flake-utils";
-    naersk = {
-      url = "github:nix-community/naersk";
+    crane = {
+      url = "github:ipetkov/crane";
       inputs.nixpkgs.follows = "nixpkgs";
     };
 
     nickel = {
       url = "github:tweag/nickel";
-      inputs = {
-        nixpkgs.follows = "nixpkgs";
-        flake-utils.follows = "utils";
-      };
+      inputs.nixpkgs.follows = "nixpkgs";
     };
 
     topiary.url = "github:tweag/topiary";
   };
 
-  outputs = { self, nixpkgs, utils, naersk, nickel, ... }@inputs: {
-    overlays = rec {
-      expects-naersk = final: _: {
-        json-schema-to-nickel = final.naersk.buildPackage {
-          pname = "json-schema-to-nickel";
-          root = ./.;
-        };
-      };
-
-      default = final: _: {
-        inherit (final.appendOverlays [
-          naersk.overlay
-          expects-naersk
-        ]) json-schema-to-nickel;
-      };
-    };
-  } // utils.lib.eachDefaultSystem (system: with import nixpkgs
-    { overlays = [ self.overlays.default ]; inherit system; }; {
-    packages.default = json-schema-to-nickel;
-
-    devShells.default = mkShell {
-      packages = [
-        cargo
-        cargo-watch
-        clippy
-        inputs.topiary.packages.${system}.default
-        nickel.packages.${system}.default
-        rust-analyzer
-        rustc
-        rustfmt
+  outputs = inputs:
+    let
+      SYSTEMS = [
+        "aarch64-darwin"
+        "aarch64-linux"
+        "x86_64-darwin"
+        "x86_64-linux"
       ];
-    };
-  });
+
+      lib = inputs.nixpkgs.lib;
+      foreach = xs: f: with lib; foldr recursiveUpdate { } (map f xs);
+      forSystems = systems: f: foreach systems (system: f system inputs.nixpkgs.legacyPackages.${system});
+    in
+    forSystems SYSTEMS (system: pkgs:
```

Comment:
any reason not to use flake-utils for that?

Reply:
Originally, I wanted outputs for all system configurations that appear in `nixpkgs.legacyPackages`. Unfortunately, some of them seem to not support Rust bootstrapping and fail during evaluation. Since `nix flake check` tries to at least evaluate all attributes for all exposed systems, that breaks the test workflow. And then the code was already there to iterate over explicitly specified systems without needing flake-utils.